command/kv_get.go

```diff
@@ -0,0 +1,136 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+var _ cli.Command = (*KVGetCommand)(nil)
+var _ cli.CommandAutocomplete = (*KVGetCommand)(nil)
+
+type KVGetCommand struct {
+	*BaseCommand
+
+	flagVersion int
+}
+
+func (c *KVGetCommand) Synopsis() string {
+	return "Retrieves data from the KV store"
+}
+
+func (c *KVGetCommand) Help() string {
+	helpText := `
+Usage: vault kv get [options] KEY
+
+  Retrieves the value from Vault's key-value store at the given key name. If no
+  key exists with that name, an error is returned. If a key exists with that
+  name but has no data, nothing is returned.
+
+      $ vault kv get secret/foo
+
+  To view the given key name at a specific version in time, specify the "-version"
+  flag:
+
+      $ vault kv get -version=1 secret/foo
+
+  Additional flags and more advanced use cases are detailed below.
+
+` + c.Flags().Help()
+	return strings.TrimSpace(helpText)
+}
+
+func (c *KVGetCommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP | FlagSetOutputFormat)
+
+	// Common Options
+	f := set.NewFlagSet("Common Options")
+
+	f.IntVar(&IntVar{
+		Name:    "version",
+		Target:  &c.flagVersion,
+		Default: 0,
+		Usage:   `If passed, the value at the version number will be returned.`,
+	})
+
+	return set
+}
+
+func (c *KVGetCommand) AutocompleteArgs() complete.Predictor {
+	return nil
+}
+
+func (c *KVGetCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *KVGetCommand) Run(args []string) int {
+	f := c.Flags()
+
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	args = f.Args()
+	switch {
```

Comment:
I see many copies of this code. Consider extracting a generic helper taking min/max number args as input.

Reply:
This is in almost every command file, i might leave this enhancement for another day.